### ISSUE TITLE
Valve Index Not Recognized ("Headset not detected")

### ISSUE THREAD

**Issue by blue42u** [ValveSoftware/SteamVR-for-Linux#313]
*I'm not sure if this issue belongs here, if it has a better home somewhere else please direct me.*

**Describe the bug**
SteamVR reports "headset not detected (108)," no matter which USB 3.x port I plug the Valve Index into. The `Valve HMD` entries also do not appear in `lsusb`, but I can access the cameras using a webcam program ([Cheese](https://wiki.gnome.org/Apps/Cheese)).

![Screenshot from 2020-03-18 18-24-23](https://user-images.githubusercontent.com/17242663/77016470-b8a98a00-6945-11ea-9ff5-b3e1a561b0da.png)
![2020-03-18-183114](https://user-images.githubusercontent.com/17242663/77016782-b4ca3780-6946-11ea-8b6d-9c63ea3cc0b4.jpg)

**To Reproduce**
1. Plug in the Index headset.
2. Start up SteamVR.
3. See error.

**Expected behavior**
A green headset icon in SteamVR.

**System Information (please complete the following information):**
- Distribution: Debian Unstable
- SteamVR version: 1.10.30
- Steam client version: Mar 18, 2020
- Opted into Steam client beta?: Yes
- Graphics driver version:  440.64 (NVIDIA)
- Gist for SteamVR System Information: [here](https://gist.github.com/blue42u/4ccb9810dbd9c085c46b4e00dcbf021c). Also [Steam's system report](https://gist.github.com/blue42u/3cd05179bf6840315c6a45acf4bf3ce8).

**Additional context**
I've been able to run The Lab by borrowing a friend's laptop and sharing the USB connection (via `usbip`) over a Wifi connection, so the Index hardware isn't faulty. It just doesn't work plugged into my gaming rig directly.

Hardware:
 - [Asus ROG Strix X570-F Gaming Motherboard](https://www.newegg.com/asus-rog-strix-x570-f-gaming/p/N82E16813119195?Item=N82E16813119195)
 - [Vantec USB 3.0 PCIe Card](https://www.newegg.com/vantec-model-ugt-pc345-pci-express-to-usb-card/p/N82E16815287019)
 - [SIIG USB 3.0 PCIe Card](https://www.newegg.com/p/15J-00S9-00002)
```
$ lspci | grep USB
05:00.0 USB controller: VIA Technologies, Inc. VL805 USB 3.0 Host Controller (rev 01)
07:00.1 USB controller: Advanced Micro Devices, Inc. [AMD] Matisse USB 3.0 Host Controller
07:00.3 USB controller: Advanced Micro Devices, Inc. [AMD] Matisse USB 3.0 Host Controller
0a:00.2 USB controller: NVIDIA Corporation TU104 USB 3.1 Host Controller (rev a1)
0a:00.3 Serial bus controller [0c80]: NVIDIA Corporation TU104 USB Type-C UCSI Controller (rev a1)
0b:00.0 USB controller: Renesas Technology Corp. uPD720201 USB 3.0 Host Controller (rev 03)
0d:00.3 USB controller: Advanced Micro Devices, Inc. [AMD] Matisse USB 3.0 Host Controller
```
The three `AMD` controllers are from my motherboard, the `Renesas` controller from the Vantec card, and the `VIA` controller from the SIIG card. None of them work with the Index, all of them work with other devices.

**Note:** Commenters who are also experiencing this issue are encouraged to include the "System Information" section in their replies.

**Comment by kisak-valve** [comment]
Hello @blue42u, can you check if you have the `steam-devices` package installed with something like `apt policy steam-devices` and install it if it isn't already?

**Comment by blue42u** [comment]
Can confirm that `steam-devices` is installed, and the installed udev rules match the most recent:
```
$ apt policy steam-devices
steam-devices:
  Installed: 1.0.0.61-2
  Candidate: 1.0.0.61-2
  Version table:
 *** 1.0.0.61-2 500
        500 http://ftp.us.debian.org/debian sid/non-free amd64 Packages
        500 http://ftp.us.debian.org/debian sid/non-free i386 Packages
        100 /var/lib/dpkg/status
$ diff -s 60-steam-vr.rules /lib/udev/rules.d/60-steam-vr.rules 
Files 60-steam-vr.rules and /lib/udev/rules.d/60-steam-vr.rules are identical
```

**Comment by ZarathustraDK** [comment]
I had this issue before. From what I gather it has something to do with your GPU's port-priority. My guess is your card has 3 displayports and you've put the Index in the "wrong" one. For me (2080TI) I had one port for my monitor, one port for my AV-hub (which is normally off), and one port for the Index. Monitor worked fine, but the index refused to get detected. Then I turned on the AV-hub, lo and behold, the index got detected.

TLDR: Try change up your displayport-inputs. There's a port 1, port 2 and a port 3 (not labelled but it's in the manual I think). Port 1 needs to be your monitor, port 2 the index OR port 3 the index as long as port 2 is also active (like a second monitor for instance).

There's probably a smart, tuxy way to fix this with a terminal, but I wouldn't know how.

**Comment by blue42u** [comment]
I've been testing without the DisplayPort, just tried with plugged in. No difference for any of the 3 DP ports. (Using HDMI for my monitor for now.)

Mind you, the displays (and everything else) work after sharing the USB from my friend's laptop. I'm starting to wonder if there's a subtle break in my tether, if that's even possible.

**Comment by kb173** [comment]
I have this issue as well. I have a dual-boot with Windows and Ubuntu set up. At first, the Index worked on Ubuntu. Later, I did a firmware update for the Index on Windows. Since then, it is not detected on Ubuntu. (It is still working fine on Windows.)

My udev rules match the most recent and I tried all DisplayPort connections. I tried Beta and default Steam. xrandr reports no connections other than the monitor. The logs say `Returning HmdNotFound because the client-side presence check failed.`.

**Comment by ZarathustraDK** [comment]
Same here, worked a week ago and has been working for weeks without a
hitch. SteamVR-launcher now gone from steam too so completely locked out.
Tried launching SteamVR from the tray-icon and this resets (!) my
KWin-session so I have to log in again. I think they messed something up
big time, hopefully they'll roll it back or rectify it like hamsters on
caffeine.

Den fre. 20. mar. 2020 kl. 12.02 skrev Karl <notifications@github.com>:

**Comment by ZarathustraDK** [comment]
Nevermind, Steam is down. Thanks Corona Chan, good luck Corona Chan. ;)

Den fre. 20. mar. 2020 kl. 22.30 skrev Henrik Peytz <henrik.peytz@gmail.com

**Comment by jug007** [comment]
For me the Index has only worked in the past if I had it plugged in before boot. Just restarting after getting the "Headset not detected" always fixed it for me.
Just adding this as it is somewhat weird on linux to actually need a restart for something to work =|

**Comment by KindDragon** [comment]
Same issue for me with Valve Index and SteamVR 1.10.32 - "Please plug in your VR headset"
https://gist.github.com/KindDragon/62cd5941c69ea77104b9ddcaed67adb9

Also, Ubuntu reboot when I exit from SteamVR

**Comment by blue42u** [comment]
After replacing my tether (Steam Support was kind enough to send me a new one), it seems to work consistently now. There are still a few software bugs (like the system buttons not always opening the dashboard), but I can play VR games now so I'm happy.

For anyone else running into this issue, try plugging into a laptop or another machine and checking if that works (with `lsusb` or equivalent), or requesting a replacement tether. It might be a hardware issue.

Closing for now, as my particular issue is resolved.

**Comment by All-The-Foxes** [comment]
I used it fine last night. Today I open Steam and the Steam client has an update, and now I can't get my headset to work no matter what I do.
